### ISSUE TITLE
1.34 npm:specifiers, Deno has panicked Error.

### ISSUE THREAD

**Issue by thyldrm** [denoland/deno#19251]
Deno 1.34 has been released and deno compile now supports npm packages.

[Deno 1.34 Updates](https://deno.com/blog/v1.34) I am doing the following based on the example in this document.

```
$ cat main.ts
import { say } from "npm:cowsay@1.5.0";
console.log(say({ text: "Hello from Deno!" }));

$ deno compile --allow-read main.ts
$ ./main.ts
```
But I am getting an error as below.

```
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.34.0
Args: ["deno", "compile", "--allow-read", "main.ts"]

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: StripPrefixError(())', cli/standalone/virtual_fs.rs:154:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

What is the reason of this ? What should we do ?

**Comment by gaui** [comment]
Same error here.

```
$ cat foo.ts 
import { say } from "npm:cowsay@1.5.0";
console.log(say({ text: "Hello from Deno!" }));
$ sudo deno compile --allow-all foo.ts ^C
$ deno -V
deno 1.34.0
$ sudo deno compile --allow-all foo.ts 
Compile file:///home/gaui/test/foo.ts to foo

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: linux x86_64
Version: 1.34.0
Args: ["deno", "compile", "--allow-all", "foo.ts"]

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: StripPrefixError(())', test/standalone/virtual_fs.rs:154:53
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by JimDabell** [comment]
This was happening for me reliably. I had an existing project using Node and I attempted to use Deno to compile it. What I discovered was that if I had already run `npm install` to populate `node_modules/` and then ran `deno compile`, it would panic as above. If I wiped the `node_modules/` directory and ran `deno compile` again, it would succeed. So it seems the trigger for this bug relates to pre-existing `node_modules/` populated by something other than Deno.

**Comment by bartlomieju** [comment]
@JimDabell we're trying to debug this error - could you please upgrade to canary version (`deno upgrade --canary`) and try to reproduce this problem again?

**Comment by JimDabell** [comment]
@bartlomieju I tried to do that when I saw the linked pull request, but canary builds aren’t available for Apple silicon. I built it from source (`25cbd97`) and saw this:

```
============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos aarch64
Version: 1.34.0
Args: ["../deno/target/debug/deno", "compile", "--allow-read", "--allow-env", "--allow-net", "--allow-all", "./index.js"]

thread 'main' panicked at 'Failed to strip prefix '/Users/disciple/Documents/Projects/dit/node_modules' from '../btoa/bin/btoa.js': prefix not found', cli/standalone/virtual_fs.rs:182:9
stack backtrace:
   0:        0x10644f61c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf92d0b9100685592
   1:        0x106477420 - core::fmt::write::hba950d59ddceeaaa
   2:        0x10643d3a4 - std::io::Write::write_fmt::h9e52517f23e48360
   3:        0x10644f42c - std::sys_common::backtrace::print::hd205fa5e4b881c8b
   4:        0x10643e0d8 - std::panicking::default_hook::{{closure}}::h1fc0ab063b55fdaf
   5:        0x10643dd9c - std::panicking::default_hook::h68f99660ab2cf986
   6:        0x101112ee4 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::h544ff9696338d930
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/alloc/src/boxed.rs:2001:9
   7:        0x100cfbce4 - deno::setup_panic_hook::{{closure}}::h8ce8dea548156d6c
                               at /Users/disciple/Documents/Projects/deno/cli/main.rs:230:5
   8:        0x10643ea4c - std::panicking::rust_panic_with_hook::haf8b9b224847163e
   9:        0x106450124 - std::panicking::begin_panic_handler::{{closure}}::h71de5f2fd48a9ed8
  10:        0x10644f714 - std::sys_common::backtrace::__rust_end_short_backtrace::h95cd4256776145cf
  11:        0x10643e5bc - _rust_begin_unwind
  12:        0x10646aa80 - core::panicking::panic_fmt::h03053be4fed319aa
  13:        0x1014ab680 - deno::standalone::virtual_fs::VfsBuilder::path_relative_root::h51db137020ab36b5
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/virtual_fs.rs:182:9
  14:        0x1014ab010 - deno::standalone::virtual_fs::VfsBuilder::add_symlink::h6dc45fe21f7189cf
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/virtual_fs.rs:154:16
  15:        0x1014aa7f8 - deno::standalone::virtual_fs::VfsBuilder::add_dir_recursive::h5d0d7b33b10a5c87
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/virtual_fs.rs:79:9
  16:        0x1014aa494 - deno::standalone::virtual_fs::VfsBuilder::add_dir_recursive::h5d0d7b33b10a5c87
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/virtual_fs.rs:71:9
  17:        0x10107290c - deno::standalone::binary::DenoCompileBinaryWriter::build_vfs::h3fbc1bccb6e0b110
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/binary.rs:540:7
  18:        0x1019685b0 - deno::standalone::binary::DenoCompileBinaryWriter::write_standalone_binary::{{closure}}::hf1e3b8bb8ccb1e1c
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/binary.rs:497:33
  19:        0x101966598 - deno::standalone::binary::DenoCompileBinaryWriter::write_bin::{{closure}}::h5f8009f1120ed8aa
                               at /Users/disciple/Documents/Projects/deno/cli/standalone/binary.rs:400:7
  20:        0x100f57650 - deno::tools::compile::compile::{{closure}}::ha4ba013378d83144
                               at /Users/disciple/Documents/Projects/deno/cli/tools/compile.rs:72:5
  21:        0x100cf867c - deno::run_subcommand::{{closure}}::{{closure}}::h922e67f54b36d5d8
                               at /Users/disciple/Documents/Projects/deno/cli/main.rs:123:52
  22:        0x101a30b10 - <futures_util::future::future::map::Map<Fut,F> as core::future::future::Future>::poll::hbecbbe32d2e640d4
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.28/src/future/future/map.rs:55:37
  23:        0x1015a4854 - <futures_util::future::future::Map<Fut,F> as core::future::future::Future>::poll::h4e01d4af9af469f5
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-util-0.3.28/src/lib.rs:91:13
  24:        0x100d39a4c - <deno_core::task::MaskFutureAsSend<F> as core::future::future::Future>::poll::h7599deb4a64d08d5
                               at /Users/disciple/Documents/Projects/deno/core/task.rs:129:13
  25:        0x100c5c230 - <core::pin::Pin<P> as core::future::future::Future>::poll::h3de2d3190521c493
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/core/src/future/future.rs:125:9
  26:        0x1019dc9d0 - tokio::runtime::task::core::Core<T,S>::poll::{{closure}}::h3a2f9d1f9665475c
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/core.rs:223:17
  27:        0x1016f5bb8 - tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut::hdd4fd54ba1d17b9e
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/loom/std/unsafe_cell.rs:14:9
  28:        0x1019d370c - tokio::runtime::task::core::Core<T,S>::poll::h65fc87cddc8515dd
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/core.rs:212:13
  29:        0x10139aa28 - tokio::runtime::task::harness::poll_future::{{closure}}::h539af9229354805a
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/harness.rs:476:19
  30:        0x1016fda44 - <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::h1993d25ea84118e0
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/core/src/panic/unwind_safe.rs:271:9
  31:        0x10120d49c - std::panicking::try::do_call::h665df9674ed0d23c
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/panicking.rs:487:40
  32:        0x101243ee8 - ___rust_try
  33:        0x1011f3614 - std::panicking::try::hb6069d7144be9a00
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/panicking.rs:451:19
  34:        0x10146d1f0 - std::panic::catch_unwind::hcd0807f8740259e1
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/panic.rs:140:14
  35:        0x101393400 - tokio::runtime::task::harness::poll_future::he4da9bdd6bc297ae
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/harness.rs:464:18
  36:        0x1013b365c - tokio::runtime::task::harness::Harness<T,S>::poll_inner::hc88e50067a9479e7
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/harness.rs:198:27
  37:        0x1013db178 - tokio::runtime::task::harness::Harness<T,S>::poll::hc5edf26db13bdcbf
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/harness.rs:152:15
  38:        0x100d0fe18 - tokio::runtime::task::raw::poll::h01fddd438bf80705
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/raw.rs:255:5
  39:        0x105281060 - tokio::runtime::task::raw::RawTask::poll::he41a10ef4a5e3cc7
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/raw.rs:200:18
  40:        0x10408a9cc - tokio::runtime::task::LocalNotified<S>::run::h66b8ee0b1071cc42
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/task/mod.rs:394:9
  41:        0x1019302d0 - tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::{{closure}}::h00b7400385815bc7
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:584:25
  42:        0x10192c914 - tokio::runtime::coop::with_budget::h9ad812dd8450e26a
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/coop.rs:107:5
  43:        0x10192c914 - tokio::runtime::coop::budget::hc6ee60874ed63b81
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/coop.rs:73:5
  44:        0x10192c914 - tokio::runtime::scheduler::current_thread::Context::run_task::{{closure}}::h7851fdde53784df4
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:285:29
  45:        0x10192c220 - tokio::runtime::scheduler::current_thread::Context::enter::hfb48455a1f21acd1
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:350:19
  46:        0x10192c5dc - tokio::runtime::scheduler::current_thread::Context::run_task::h7be884a4aac14eab
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:285:9
  47:        0x10192ebbc - tokio::runtime::scheduler::current_thread::CoreGuard::block_on::{{closure}}::h1f82a36b881fac03
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:583:34
  48:        0x10192dba4 - tokio::runtime::scheduler::current_thread::CoreGuard::enter::{{closure}}::h01a6d4396a039da8
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:615:57
  49:        0x1016cdc1c - tokio::macros::scoped_tls::ScopedKey<T>::set::hdebcc3a2d6921397
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/macros/scoped_tls.rs:61:9
  50:        0x10192d720 - tokio::runtime::scheduler::current_thread::CoreGuard::enter::hb73792b4f339adb9
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:615:27
  51:        0x10192dd60 - tokio::runtime::scheduler::current_thread::CoreGuard::block_on::h88a19b8272e9c577
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:530:19
  52:        0x10191cb1c - tokio::runtime::scheduler::current_thread::CurrentThread::block_on::h4b5d159ff2408d5f
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/scheduler/current_thread.rs:154:24
  53:        0x101b4da98 - tokio::runtime::runtime::Runtime::block_on::h0e55d814a218a490
                               at /Users/disciple/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.28.1/src/runtime/runtime.rs:302:47
  54:        0x101ad4738 - deno_runtime::tokio_util::create_and_run_current_thread::h03b9de45353a9bdc
                               at /Users/disciple/Documents/Projects/deno/runtime/tokio_util.rs:70:3
  55:        0x101ad4738 - deno::main::h84ab4b781c624a64
                               at /Users/disciple/Documents/Projects/deno/cli/main.rs:311:34
  56:        0x10126e89c - core::ops::function::FnOnce::call_once::h7f566ab56afc8a17
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/core/src/ops/function.rs:250:5
  57:        0x1015af030 - std::sys_common::backtrace::__rust_begin_short_backtrace::h5ec740d9aacadccf
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/sys_common/backtrace.rs:134:18
  58:        0x100c5e614 - std::rt::lang_start::{{closure}}::hd1b6e70b23e90240
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/rt.rs:166:18
  59:        0x10643cc0c - std::rt::lang_start_internal::hcf5249aa241f98ab
  60:        0x100c5e5e0 - std::rt::lang_start::h16a2b76a8b4ce996
                               at /private/tmp/rust-20230424-7223-163y6w3/rustc-1.69.0-src/library/std/src/rt.rs:165:17
  61:        0x101ad4b68 - _main
```

**Comment by bartlomieju** [comment]
Thanks, that's very useful and we should be able to fix the issue with this data! Much appreciated.

**Comment by thyldrm** [comment]
I'm writing for more information. After upgrading to the Canary version, I encountered an output like the one below.
Note: Mac with Intel processor. It's not silicone.

```

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.34.0+e95f098
Args: ["deno", "compile", "--allow-read", "main.ts"]

thread 'main' panicked at 'Failed to strip prefix '/Users/cyberwise/node_modules' from '../../../update-browserslist-db/cli.js': prefix not found', cli/standalone/virtual_fs.rs:182:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Lordmau5** [comment]
I ran into this same issue just now as well, not sure if this is any different from above

```E:\Development\AlwaysOnTV\bundle>deno compile --import-map=import_map.json index.js
Compile file:///E:/Development/AlwaysOnTV/lkjdfglkdjfgg/index.js to lkjdfglkdjfgg.exe

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: windows x86_64
Version: 1.34.0+e95f098
Args: ["deno", "compile", "--import-map=import_map.json", "index.js"]

thread 'main' panicked at 'Failed to strip prefix 'E:\Development\AlwaysOnTV\bundle\node_modules' from 'E:\Development\AlwaysOnTV\bundle\src\': prefix not found', cli\standalone\virtual_fs.rs:182:9
stack backtrace:
   0:     0x7ff771ccc489 - napi_release_threadsafe_function
   1:     0x7ff770e2dedb - napi_release_threadsafe_function
   2:     0x7ff771ca685a - napi_release_threadsafe_function
   3:     0x7ff771cd0b9b - napi_release_threadsafe_function
   4:     0x7ff771cd0791 - napi_release_threadsafe_function
   5:     0x7ff771cd03a5 - napi_release_threadsafe_function
   6:     0x7ff770c6947c - napi_fatal_error
   7:     0x7ff771cd13e7 - napi_release_threadsafe_function
   8:     0x7ff771cd1156 - napi_release_threadsafe_function
   9:     0x7ff771cd109f - napi_release_threadsafe_function
  10:     0x7ff771cd1074 - napi_release_threadsafe_function
  11:     0x7ff772f34085 - CrashForExceptionInNonABICompliantCodeRange
  12:     0x7ff770bbc89f - napi_fatal_error
  13:     0x7ff770bbb5b0 - napi_fatal_error
  14:     0x7ff770bb6a31 - napi_fatal_error
  15:     0x7ff770bd5956 - napi_fatal_error
  16:     0x7ff7709da258 - napi_cancel_async_work
  17:     0x7ff7708d6688 - napi_cancel_async_work
  18:     0x7ff770c6b00c - napi_fatal_error
  19:     0x7ff770803566 - <unknown>
  20:     0x7ff770d50913 - napi_release_threadsafe_function
  21:     0x7ff772ee6ecc - CrashForExceptionInNonABICompliantCodeRange
  22:     0x7ffc6bd726ad - BaseThreadInitThunk
  23:     0x7ffc6cb4a9f8 - RtlUserThreadStart```

It compiles just fine if the node_modules folder doesn't exist, however it brings up an error with better-sqlite3 when I try to launch it (which seems to be discussed in another issue)

**Comment by jpaquim** [comment]
Getting the same when trying out the vite example from the blog post:
```
Platform: macos aarch64
Version: 1.34.0
Args: ["deno", "compile", "--allow-read", "--allow-write", "--allow-env", "--allow-net", "npm:vite"]

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: StripPrefixError(())', cli/standalone/virtual_fs.rs:154:53
stack backtrace:
   0:        0x10367632c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf92d0b9100685592
   1:        0x102e533fc - core::fmt::write::hba950d59ddceeaaa
   2:        0x103667a3c - std::io::Write::write_fmt::h9e52517f23e48360
   3:        0x103676164 - std::sys_common::backtrace::print::hd205fa5e4b881c8b
   4:        0x103668610 - std::panicking::default_hook::{{closure}}::h1fc0ab063b55fdaf
   5:        0x103668310 - std::panicking::default_hook::h68f99660ab2cf986
   6:        0x102dbedd4 - deno::setup_panic_hook::{{closure}}::hf81e38847c187047
   7:        0x103668f1c - std::panicking::rust_panic_with_hook::haf8b9b224847163e
   8:        0x103676658 - std::panicking::begin_panic_handler::{{closure}}::h71de5f2fd48a9ed8
   9:        0x1036765c8 - std::sys_common::backtrace::__rust_end_short_backtrace::h95cd4256776145cf
  10:        0x103668aa0 - _rust_begin_unwind
  11:        0x102e48e28 - core::panicking::panic_fmt::h03053be4fed319aa
  12:        0x102e4a0e4 - core::result::unwrap_failed::hbab326b6fd47affb
  13:        0x102bcdaf4 - core::result::Result<T,E>::unwrap::h001914f0b79b2779
  14:        0x102bcc420 - deno::standalone::virtual_fs::VfsBuilder::add_dir_recursive::h687cbfe5c501af62
  15:        0x102bcc200 - deno::standalone::virtual_fs::VfsBuilder::add_dir_recursive::h687cbfe5c501af62
  16:        0x102bc9850 - deno::standalone::binary::DenoCompileBinaryWriter::write_standalone_binary::{{closure}}::hb9a5a0e5b27b52fa
  17:        0x102bc7580 - deno::tools::compile::compile::{{closure}}::h87c251e88dd1c538
  18:        0x102bc5710 - <deno_core::task::MaskFutureAsSend<F> as core::future::future::Future>::poll::h15ee7068006f2ec3
  19:        0x102bce20c - tokio::runtime::task::raw::poll::h954a695cc2054e3c
  20:        0x102afc24c - deno::main::h8f5200e14e2db6f3
  21:        0x102aee7fc - std::sys_common::backtrace::__rust_begin_short_backtrace::hec0983caf118bb9d
  22:        0x102afb190 - _main
```

**Comment by JimDabell** [comment]
I’ve just tried #19285 and it resolves this panic for me @dsherret.

**Comment by gaui** [comment]
When will this fix be released?

**Comment by JimDabell** [comment]
@gaui It's already released, the fix is in v1.34.1.
